ng-dev/release/notes/changelog.ts

```diff
@@ -0,0 +1,127 @@
+import {existsSync, readFileSync, writeFileSync} from 'fs';
+import {join} from 'path';
+import * as semver from 'semver';
+import {GitClient} from '../../utils/git/git-client';
+
+/** Project-relative path for the changelog file. */
+export const changelogPath = 'CHANGELOG.md';
+
+/** Project-relative path for the changelog archive file. */
+export const changelogArchivePath = 'CHANGELOG_ARCHIVE.md';
+
+/** A marker used to split a CHANGELOG.md file into individual entries. */
+export const splitMarker = '<!-- CHANGELOG SPLIT MARKER -->';
+
+/**
+ * A string to use between each changelog entry when joining them together.
+ *
+ * Since all every changelog entry's content is trimmed, when joining back together, two new lines
+ * must be placed around the splitMarker to create a one line buffer around the comment in the
+ * markdown.
+ * i.e.
+ * <changelog entry content>
+ *
+ * <!-- CHANGELOG SPLIT MARKER -->
+ *
+ * <changelog entry content>
+ */
+const joinMarker = `\n\n${splitMarker}\n\n`;
+
+/** A RegExp matcher to extract the version of a changelog entry from the entry content. */
+const versionAnchorMatcher = new RegExp(`<a name="(.*)"></a>`);
+
+/** An individual changelog entry. */
+interface ChangelogEntry {
+  content: string;
+  version: semver.SemVer;
+}
+
+export class Changelog {
+  /** The absolute path to the changelog file. */
+  readonly changelogPath = join(this.git.baseDir, changelogPath);
+  /** The absolute path to the changelog archive file. */
+  readonly changelogArchivePath = join(this.git.baseDir, changelogArchivePath);
+
+  /** The changelog entries in the CHANGELOG.md file. */
+  private entries = this.getEntriesFor(this.changelogPath);
+  /** The changelog entries in the CHANGELOG_ARCHIVE.md file. */
+  private archiveEntries = this.getEntriesFor(this.changelogArchivePath);
```

Comment:
IIRC there were some problems with reading _such_ large files in a Node process. I think we should only read files when actually needed?